units/generate_tasks_test.go

```diff
@@ -266,6 +431,128 @@ func TestGenerateTasks(t *testing.T) {
 	assert.Equal("mci_identifier_race_detector_display_my_display_task__01_01_01_00_00_00", b.Tasks[0].Id)
 }
 
+func TestGeneratedTasksAreNotDependencies(t *testing.T) {
+	assert := assert.New(t)
+	require := require.New(t)
+	require.NoError(db.ClearCollections(model.ProjectRefCollection, model.VersionCollection, build.Collection, task.Collection, distro.Collection, patch.Collection, model.ParserProjectCollection))
+	defer require.NoError(db.ClearCollections(model.ProjectRefCollection, model.VersionCollection, build.Collection, task.Collection, distro.Collection, patch.Collection, model.ParserProjectCollection))
+	v := model.Version{
+		Id:         "sample_version",
+		Identifier: "mci",
+		BuildVariants: []model.VersionBuildStatus{{
+			BuildVariant: "generate-tasks-for-version",
+			BuildId:      "b1",
+		}, {
+			BuildVariant: "testBV1",
+			BuildId:      "b2",
+		}, {
+			BuildVariant: "testBV2",
+			BuildId:      "b3",
+		}},
+	}
+	require.NoError(v.Insert())
+	b1 := build.Build{
+		Id:           "b1",
+		BuildVariant: "generate-tasks-for-version",
+		Version:      "sample_version",
+	}
+	b2 := build.Build{
+		Id:           "b2",
+		BuildVariant: "testBV1",
+		Version:      "sample_version",
+	}
+	b3 := build.Build{
+		Id:           "b3",
+		BuildVariant: "testBV2",
+		Version:      "sample_version",
+	}
+	require.NoError(b1.Insert())
+	require.NoError(b2.Insert())
+	require.NoError(b3.Insert())
+
+	pp := model.ParserProject{}
+	err := util.UnmarshalYAMLWithFallback([]byte(dependOnGeneratedTasksDisabledConfig), &pp)
+	require.NoError(err)
+	pp.Id = "sample_version"
+	require.NoError(pp.Insert())
+	generateTask := task.Task{
+		Id:                    "mci_identifier_generate_tasks_for_version_version_gen__01_01_01_00_00_00",
+		Version:               "sample_version",
+		BuildId:               "b1",
+		Project:               "mci",
+		DisplayName:           "version_gen",
+		GeneratedJSONAsString: sampleGeneratedProject2,
+		Status:                evergreen.TaskStarted,
+	}
+	require.NoError(generateTask.Insert())
+	projectRef := model.ProjectRef{Id: "mci", Identifier: "mci_identifier"}
+	require.NoError(projectRef.Insert())
+
+	j := NewGenerateTasksJob(generateTask.Version, generateTask.Id, "1")
+	j.Run(context.Background())
+	assert.NoError(j.Error())
+	tasks := []task.Task{}
+	assert.NoError(db.FindAllQ(task.Collection, db.Query(task.ByVersion("sample_version")), &tasks))
+	assert.Len(tasks, 4)
+	for _, foundTask := range tasks {
+		switch foundTask.DisplayName {
+		case "version_gen":
```

Comment:
nit: I think you can do `case "version_gen", "dependency_task":` since checking the same thing